library/src/main/java/io/github/wax911/library/annotation/processor/fragment/FragmentPatcher.kt

```diff
@@ -0,0 +1,58 @@
+package io.github.wax911.library.annotation.processor.fragment
+
+import android.util.Log
+
+/**
+ * This class will return a String containing fragment definitions. This String can be appended to a query in order to
+ * have all referenced fragments fully defined, before sending it to the server. This allows a query (for example) to
+ * reference fragments, but have the definition of those fragments live outside of the query file.
+ *
+ * This class also properly handles recursion. This means fragments can have references to other fragments, and it will
+ * do its best to resolve everything correctly.
+ */
+class FragmentPatcher(
```

Comment:
The new class that tries to find the externally defined fragments, and pieces them together.